network-config-analyzer/GenericTreeScanner.py

```diff
@@ -0,0 +1,95 @@
+#
+# Copyright 2020- IBM Inc. All rights reserved
+# SPDX-License-Identifier: Apache2.0
+#
+
+import os
+import abc
+from enum import Enum
+from dataclasses import dataclass
+from ruamel.yaml import YAML, error
+from sys import stderr
+
+
+@dataclass
+class YamlFile:
+    """
+    A class for holding the retrieved data of a yaml file from git repo
+    """
+    data: object
+    path: str
+
+
+class GenericTreeScanner(metaclass=abc.ABCMeta):
```

Comment:
```suggestion
class GenericTreeScanner(abc.ABC):
```
Although both forms should work, I think this is the preferred form today. See https://docs.python.org/3/library/abc.html